windows/src/desktop/kmshell/install/UfrmInstallKeyboardFromWeb.pas

```diff
@@ -117,7 +118,17 @@ procedure TfrmInstallKeyboardFromWeb.TntFormCreate(Sender: TObject);
 
 procedure TfrmInstallKeyboardFromWeb.cefBeforeBrowseSync(Sender: TObject; const Url: string; out Handled: Boolean);
 begin
-  Handled := TRegEx.IsMatch(Url, UrlPath_RegEx_MatchKeyboardsInstall);
+  // This introduces some deeper knowledge of URL paths in Keyman Configuration, which is
+  // a bit of a shame, because we try to keep our internal knowledge to /go/ urls on
+  // keyman.com. However, there is not really any great way around this that I've found,
+  // which allows us to handle internal navigation on the keyboard search and still lets
+  // us open other URLs that may be in the search results in an external browser.
+  Handled :=
+    TRegEx.IsMatch(Url, URLPath_RegEx_MatchKeyboardsInstall) or       // capture https://keyman.com/keyboards/install/*
+    (not TRegEx.IsMatch(Url, UrlPath_RegEx_MatchKeyboardsRoot) and    // don't capture https://keyman.com/keyboards*
```

Comment:
I'm a little confused why we don't capture https://keyman.com/keyboards*

e.g. https://keyman.com/keyboards/khmer_angkor

Shouldn't we treat as an internal link and handle it?

Reply:
"handle" as in render it in the embedded keyboard search instead of kicking to external browser

Reply:
Handle means that we are capturing the URL and not letting it go through to the embedded browser; it is then processed in the `cefBeforeBrowse` function (`cefBeforeBrowseSync` happens in the CEF thread but we have to do the actual processing in the main thread).